Homeworks/04/src/components/Fetch.tsx

```diff
@@ -10,26 +10,43 @@ interface PokemonResponse {
 
 export default function Fetch() {
   const [pokemon, setPokemon] = useState<PokemonResponse[]>([])
+  const [id, setId] = useState<number>(1)
 
   // your task here is to fetch Pokémon from a given URL and just display its name
   // first Pokémon should be fetched when the compontent is first rendered
   // then, once user clicks on FETCH button, increment your id to fetch the next one and add it to the list
 
+  useEffect(() => {
+    fetchPokemon(id).then(res => {
+      setPokemon(pokemon => [...pokemon, res])
+    })
+  }, [])
+
   // fetch works just as it worked in plain JavaScript, the only difference is you need to specify what the
   // type of the object is, after converting it to .json()
   // e.g. code snippet
   const fetchPokemon = async (id: number) => {
     //we fetch the Pokemon, convert it to json and we have p object of type PokemonResponse
     const p: PokemonResponse = await (await fetch(getPokemonUrl(id))).json()
+    return p
   }
 
   // return 2 elements:
   // 1st: a button which will trigger another fetch on click, for a Pokémon with next id
   // 2nd: All Pokémon stored in your list
   return (
     <div>
-      <button>FETCH</button>
-      {/* display Pokémon in a list here, just display a div with its name for each Pokémon */}
+      <button onClick={() => {
+        fetchPokemon(id+1).then(res => {
+          setPokemon(pokemon => [...pokemon, res].sort(function(a: PokemonResponse, b: PokemonResponse) {
+            return a.name.localeCompare(b.name)
+          }))
+        })
+        setId(id+1)
```

Comment:
Sa svime što sam napisao iznad, ovdje bi nam ostala samo ova linija:

```suggestion
        setId(id+1)
```

Međutim, i ovdje postoji problem a to je kada pokušavamo postaviti vrijednost varijable koja ovisi o prethodnoj vrijednosti (kada radimo increment / decrement). Kada radimo na ovaj način iznad, ukoliko imamo nekoliko slijednih poziva setState, React neće ispravno moći postaviti novu vrijednost. Iz tog razloga moramo proslijediti funkciju za update:

```suggestion
        setId(previousId => previousId +1)
```

Vidi primjer: https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state